packages/interceptors-opentelemetry/src/worker/index.ts

```diff
@@ -34,7 +34,7 @@ export class OpenTelemetryActivityInboundInterceptor implements ActivityInboundC
   }
 
   async execute(input: ActivityExecuteInput, next: Next<ActivityInboundCallsInterceptor, 'execute'>): Promise<unknown> {
-    const context = await extractContextFromHeaders(input.headers);
+    const context = await Promise.resolve(extractContextFromHeaders(input.headers));
```

Comment:
I kept the await here since we can't use the sdk flags to gate it. I don't think this promise can affect replaying, but I left it since I don't believe we can use SDK flags in this context. Happy to remove promise if you think that would be a better solution.